extractor/filesystem/os/apk/extractor.go

```diff
@@ -129,67 +126,91 @@ func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]
 	return inventory, err
 }
 
+// parseSingleApkRecord reads from the scanner a single record,
+// returns nil, nil when scanner ends.
+func parseSingleApkRecord(scanner *bufio.Scanner) (map[string]string, error) {
+	// There is currently 26 keys defined here (Under "Installed Database V2"):
+	// https://wiki.alpinelinux.org/wiki/Apk_spec
+	var group map[string]string = make(map[string]string, 30)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line != "" {
+			key, val, found := strings.Cut(line, ":")
+
+			if !found {
+				return nil, fmt.Errorf("invalid line: %q", line)
+			}
+
+			group[key] = val
+			continue
+		}
+
+		// check both that line is empty and we have filled out data in group
+		// this avoids double empty lines returning early
+		if line == "" && len(group) > 0 {
+			return group, nil
+		}
+	}
+
+	// Err() could only be non nil when Scan() returns false
```

Comment:
When you reach the end of the file. Scan() will be false and Err() will be nil, no?

Reply:
The comment is referring to why I don't return scanner.Err() on line 154, I'll move this comment up to there which should make more sense.